core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -445,6 +450,8 @@ object SparkSubmit {
         sysProp = "spark.submit.deployMode"),
       OptionAssigner(args.name, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES, sysProp = "spark.app.name"),
       OptionAssigner(args.ivyRepoPath, ALL_CLUSTER_MGRS, CLIENT, sysProp = "spark.jars.ivy"),
+      OptionAssigner(args.ivySettingsFile, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES,
```

Comment:
Is this setting needed by the application? It seems it only really affects `SparkSubmit` itself.